File: python/day-1/1_2.py
import re

inp = []
list_1 = []
list_2 = []
answer = 0

with open("input.txt", "r") as f:
  lines = f.readlines()
  for line in lines:
    inp.append(line.strip())

for i in inp:
  v = i.split("   ")
  list_1.append(v[0])
  list_2.append(v[1])

for i in range(len(list_1)):
  answer += int(list_1[i]) * list_2.count(list_1[i])
  
print(answer)

File: rust/day-1/1_2.rs
use std::fs;

fn main() {
    let file = "input.txt";
    let filename = format!("./{}", file);
    let binding = fs::read_to_string(filename).expect("Something went wrong reading the file");

    let contents = binding.trim().split("\n").collect::<Vec<&str>>();

    let mut list_1 = Vec::new();
    let mut list_2 = Vec::new();

    for i in 0..contents.len() {
        let t = contents[i].split("   ").collect::<Vec<&str>>();
        list_1.push(t[0].parse::<i32>().unwrap());
        list_2.push(t[1].parse::<i32>().unwrap());
    }

    let mut result = 0;
    for i in 0..list_1.len() {
        result += list_1[i] * list_2.iter().filter(|&x| x == &list_1[i]).count() as i32;
    }

    println!("Result: {:?}", result);
}
